src/launcher/plugin.ts

```diff
@@ -0,0 +1,290 @@
+const CWD_CLASS = 'jp-LauncherWidget-cwd';
+
+/**
+ * The class name added to LauncherWidget body nodes.
+ */
+const BODY_CLASS = 'jp-LauncherWidget-body';
+
+/**
+ * The class name added to LauncherWidget dialog node.
+ */
+const DIALOG_CLASS = 'jp-LauncherWidget-dialog';
+
+
+/**
+ * A service providing an interface to the the launcher.
+ */
+export
+const launcherExtension: JupyterLabPlugin<ILauncher> = {
+  id: 'jupyter.extensions.launcher',
+  requires: [IServiceManager, IPathTracker, ICommandPalette],
+  provides: ILauncher,
+  activate: activateLauncher,
+  autoStart: true
+};
+
+
+/**
+ * Activate the launcher.
+ */
+function activateLauncher(app: JupyterLab, services: IServiceManager, pathTracker: IPathTracker, palette: ICommandPalette): ILauncher {
+  let launcherModel = new LauncherModel();
+
+  launcherModel.setDir(pathTracker.path);
+  launcherModel.setApp(app);
+
+  pathTracker.pathChanged.connect(() => {
+    launcherModel.setDir(pathTracker.path);
+  });
+
+  let launcherWidget = new LauncherWidget()
+
+  launcherWidget.model = launcherModel;
+  launcherWidget.id = 'landing-jupyterlab-widget';
+  launcherWidget.title.label = 'Launcher';
+
+  // Hardcoded defaults.
+  let names = [
+    'Notebook',
+    'Console',
+    'Terminal',
+    'Text Editor',
+  ];
+
+  let actions = [
+    'file-operations:new-notebook',
+    `console:create-${services.kernelspecs.default}`,
+    'terminal:create-new',
+    'file-operations:new-text-file',
+  ]
+
+  app.commands.addCommand('jupyterlab-launcher:add-item', {
+    label: 'Add Launcher Item',
+    execute: (args) => {
+      launcherModel.add(args['name'] as string, args['action'] as string,
+                        args['args'] as JSONObject, args['imgName'] as string);
+    }
+  });
+
+  for (let i in names) {
+    // Note: we do not retain a handle on the items added by default, which
+    // means we have to way of removing them after the fact.
+    //launcherModel.add(names[i], actions[i]);
+    //launcherModel.add(names[i], actions[i]);
+    app.commands.execute('jupyterlab-launcher:add-item', {name: names[i], action: actions[i]});
+  }
+
+    app.commands.execute('jupyterlab-launcher:add-item', {name: 'larger font', action: 'terminal:increase-font', imgName: "jp-ImageTerminal fa fa-plus"})
+
+    app.commands.execute('jupyterlab-launcher:add-item', {name: 'smaller font', action: 'terminal:decrease-font', imgName: "jp-ImageTerminal fa fa-minus"})
+
+
+  app.commands.addCommand('jupyterlab-launcher:show', {
+    label: 'Show Launcher',
+    execute: () => {
+      if (!launcherWidget.isAttached) {
+        app.shell.addToLeftArea(launcherWidget);
+      }
+      app.shell.activateLeft(launcherWidget.id);
+    }
+  });
+
+  palette.addItem({
+    command: 'jupyterlab-launcher:show',
+    category: 'Help'
+  });
+
+  app.shell.addToLeftArea(launcherWidget);
+  return launcherModel;
+}
+
+
+/**
+ * Simple encapsulation of name and callback of launcher entries.
+ */
+class LauncherItem {
```

Comment:
In general, we need to keep most things out of the plugin.ts file so they can, at least in principle, be used outside of jupyterlab. Some options:

* Move the `Launcher` and `LauncherItem` to the same file as the interface, maybe `launcher.ts`?
* Move the model to `model.ts` and widget to `widget.ts`?
* Or just move all of those  things to the `index.ts`?

But basically the only things that should be in the plugin are the plugin declaration and activation function...